### PR TITLE
Added UPnP Support from #13132

### PR DESCRIPTION
**Pull request by hrydgard**
More from #13132 @ANR2ME .

Getting it to build on Windows UWP took some doing - had to fork miniupnpc to fix a linking error due to UWP being stupid.

Squished away a couple of tiny commits (`Fix i18n type` and `Fix Local IP detection on non-Windows system`, the latter just contained a line break).

Making a PR to get it upstream: https://github.com/miniupnp/miniupnp/pull/486

### CODE REVIEW

**Comment by hrydgard** [comment]
Argh, so many attempts...

**Comment by hrydgard** [comment]
Sigh. One thing left - figuring out how to get Android.mk to generate `ext/miniupnp/miniupnpc/miniupnpcstrings.h` . We have solutions for the other build systems but I'm really tempted to just check one in for Android...

That's for tomorrow or something though.

File: UWP/miniupnpc_UWP/miniupnpc_UWP.vcxproj.filters [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  <ItemGroup>
+    <Filter Include="src">
+      <UniqueIdentifier>{3e20a98f-6a6c-4ba1-8b5a-d44430460dd1}</UniqueIdentifier>
+    </Filter>
+    <Filter Include="scripts">
+      <UniqueIdentifier>{f940ecfb-a971-43b3-9f9d-afe3545e9798}</UniqueIdentifier>
+    </Filter>
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="..\..\ext\miniupnp\miniupnpc\msvc\genminiupnpcstrings.vbs">
```
Comment (ANR2ME) [inline UWP/miniupnpc_UWP/miniupnpc_UWP.vcxproj.filters:98]: Don't forget to run this vbs script while building to generate a header file that didn't came with miniUPnP repo :)

Comment (hrydgard) [reply UWP/miniupnpc_UWP/miniupnpc_UWP.vcxproj.filters:98]: Well yes, I know  - I had to set that up for UWP. But for Android.mk which can run on multiple platforms it's not so simple....

Comment (ANR2ME) [reply UWP/miniupnpc_UWP/miniupnpc_UWP.vcxproj.filters:98]: True, i ended copying a pre-grenerate header file for Android.mk LOL

**Comment by ANR2ME** [comment]
> 
> 
> Sigh. One thing left - figuring out how to get Android.mk to generate `ext/miniupnp/miniupnpc/miniupnpcstrings.h` . We have solutions for the other build systems but I'm really tempted to just check one in for Android...
> 
> That's for tomorrow or something though.

There is this `updateminiupnpcstrings.sh` that came with the repo, may be you can run that bash script before building?

**Comment by ANR2ME** [comment]
Btw are these commits can also be included here later? most of them are only related to `__NetInit/__NetShutdown` or PortManager.cpp/h so shouldn't cause much conflict.
https://github.com/hrydgard/ppsspp/pull/13132/commits/a3ce9050bc33628b039fc3f8fbfd5073ff77ff97
https://github.com/hrydgard/ppsspp/pull/13132/commits/e46060ed14e0a8dce21c122b811947591f2f06a1

**Comment by hrydgard** [comment]
Both of those cause some conflicts to cherry-pick due to changes in commits in between.

In general, it's good to try to do focused commits that only touch one area at a time, not just fix up little things everywhere when you see them - makes it more easy to reorder things when needed :)

**Pull request merged**